### PR TITLE
feat: replace EnsController with core implementation

### PR DESCRIPTION
**Pull request by cryptodev-2s**
## **Description**

Following the successful merge of EnsController from the extension to the core Existing EnsController as detailed in https://github.com/MetaMask/core/issues/1129, we should now fully remove the EnsController logic from the extension and instead rely entirely on the core monorepo implementation.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23224?quickstart=1)

## **Related issues**

Fixes: 23223

## Changelog

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by cryptodev-2s** [comment]
@metamaskbot update-policies

File: app/scripts/controllers/ens/ens.js [hunk 1]
```diff
@@ -1,29 +0,0 @@
-import { Web3Provider } from '@ethersproject/providers';
-import ensNetworkMap from 'ethereum-ens-network-map';
```
[the remark this answers is not in the document]
Comment (cryptodev-2s) [reply app/scripts/controllers/ens/ens.js:2]: @legobeat I see that it's still being used here https://github.com/MetaMask/metamask-extension/blob/develop/ui/ducks/domains.js

[the remark this answers is not in the document]
Comment (mcmire) [reply app/scripts/controllers/ens/ens.js:2]: Hmm, interesting. I wonder if there's some duplicate code that we need to get rid or move into EnsController. We can revisit this in a future PR.

**Review by mcmire — APPROVED** [review]
Looks good!

**Review by legobeat — APPROVED** [review]

File: app/scripts/controllers/ens/ens.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcmire) [reply app/scripts/controllers/ens/ens.js:2]: Oh I see, yes, if we inline the ENS registry map then we can bring it in from `ens-controller` for that particular use case. That makes sense.

**Pull request merged**